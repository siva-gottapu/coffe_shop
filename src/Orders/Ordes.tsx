import React from "react";
import { createStyles, Theme, makeStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import { List, ListItemText } from "@material-ui/core";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: "100%",
      backgroundColor: theme.palette.background.paper,
    },
    flexContainer: {
      display: "flex",
      flexDirection: "row",
      padding: 0,
    },
  })
);

function Orders(props: any) {
  const classes = useStyles();

  const orderList = () => {
    if (props.orders) {
      const items = props.orders.map((text: any) => {
        return (
          <ListItem button key={text}>
            <ListItemText primary={text} />
          </ListItem>
        );
      });
      return items;
    }
  };
  return (
    <div className={classes.root}>
      <List className={classes.flexContainer}>{orderList()}</List>
    </div>
  );
}

export default Orders;
