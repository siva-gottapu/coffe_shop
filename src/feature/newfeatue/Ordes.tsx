import React from "react";
import { createStyles, Theme, makeStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import { List, ListItemText } from "@material-ui/core";
import { OrdersModel } from "../../model";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: "100%",
      backgroundColor: theme.palette.background.paper,
    },
    flexContainer: {
      display: "flex",
      flexDirection: "row",
      padding: 0,
    },
  })
);

type OrdersProps = {
  orders: OrdersModel[];
};

function Orders(props: OrdersProps) {
  const classes = useStyles();

  const orderList = () => {
    if (props.orders) {
      const items = props.orders.map((order: OrdersModel) => {
        return (
          <ListItem button key={order.id}>
            <ListItemText primary={order.id} />
          </ListItem>
        );
      });
      return items;
    }
  };
  return (
    <div className={classes.root}>
      <List className={classes.flexContainer}>{orderList()}</List>
    </div>
  );
}

export default Orders;
