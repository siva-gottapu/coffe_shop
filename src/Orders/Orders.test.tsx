import React from "react";
import Enzyme from "enzyme";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Ordes from "./Ordes";
Enzyme.configure({ adapter: new Adapter() });

describe("Ordes component", () => {
  it("renders correct structure", () => {
    expect(shallow(<Ordes />)).toMatchSnapshot();
  });
});
