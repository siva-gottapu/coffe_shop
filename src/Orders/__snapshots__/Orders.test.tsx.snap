// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ordes component renders correct structure 1`] = `ShallowWrapper {}`;
